jobsdb/jobsdb.go

```diff
@@ -3385,59 +3386,31 @@ func (jd *HandleT) deleteJobStatusInTx(txHandler transactionHandler, conditions
 stateFilters and customValFilters do a OR query on values passed in array
 parameterFilters do a AND query on values included in the map
 */
-func (jd *HandleT) deleteJobStatusDSInTx(txHandler transactionHandler, ds dataSetT, conditions QueryConditions) (int, error) {
-	stateFilters := conditions.StateFilters
-	customValFilters := conditions.CustomValFilters
-	parameterFilters := conditions.ParameterFilters
-
-	checkValidJobState(jd, stateFilters)
+func (jd *HandleT) failExecutingDSInTx(txHandler transactionHandler, ds dataSetT) error {
+	queryStat := jd.getTimerStat(
+		"jobsdb_fail_executing_ds_time",
+		&statTags{CustomValFilters: []string{jd.tablePrefix}},
+	)
 
-	tags := statTags{CustomValFilters: conditions.CustomValFilters, StateFilters: conditions.StateFilters, ParameterFilters: conditions.ParameterFilters}
-	queryStat := jd.getTimerStat("delete_job_status_ds_time", &tags)
 	queryStat.Start()
 	defer queryStat.End()
 
-	var stateQuery string
-	var sqlFilters []string
-	if len(stateFilters) > 0 {
-		stateQuery = " AND " + constructQueryOR("job_state", stateFilters)
-	}
-	if conditions.AfterJobID != nil {
-		sqlFilters = append(sqlFilters, fmt.Sprintf(`job_id > %d`, *conditions.AfterJobID))
-	}
-	if len(customValFilters) > 0 {
-		sqlFilters = append(sqlFilters, constructQueryOR(fmt.Sprintf(`%q.custom_val`, ds.JobTable), customValFilters))
-	}
-	if len(parameterFilters) > 0 {
-		sqlFilters = append(sqlFilters, constructParameterJSONQuery(ds.JobTable, parameterFilters))
-	}
-
-	sqlFiltersString := strings.Join(sqlFilters, " AND ")
-	if len(sqlFiltersString) > 0 {
-		sqlFiltersString = "WHERE " + sqlFiltersString
-	}
-
-	sqlStatement := fmt.Sprintf(`DELETE FROM %[1]q
-									WHERE id = ANY(
-										SELECT id from "v_last_%[1]s" where job_id IN (SELECT job_id from %[2]q %[3]s)
-									) %[4]s AND retry_time < $1`,
-		ds.JobStatusTable, ds.JobTable, sqlFiltersString, stateQuery)
+	sqlStatement := fmt.Sprintf(
+		`UPDATE %[1]q SET job_state='failed'
+			WHERE id = ANY(
+				SELECT id from "v_last_%[1]s" where job_state='executing'
+			) AND retry_time < $1`,
+		ds.JobStatusTable,
+	)
 
 	stmt, err := txHandler.Prepare(sqlStatement)
 	if err != nil {
-		return 0, err
+		return err
 	}
 	defer func() { _ = stmt.Close() }()
-	res, err := stmt.Exec(getTimeNowFunc())
-	if err != nil {
-		return 0, err
-	}
-	deleteCount, err := res.RowsAffected()
-	if err != nil {
-		return 0, err
-	}
 
-	return int(deleteCount), nil
+	_, err = stmt.Exec(getTimeNowFunc())
+	return err
```

Comment:
it could be simplified
```
_, err := txHandler.Exec(fmt.Sprintf(
		`UPDATE %[1]q SET job_state='failed'
			WHERE id = ANY(
				SELECT id from "v_last_%[1]s" where job_state='executing'
			) AND retry_time < $1`,
		ds.JobStatusTable,
	))
	return err
```